File: src/cpp_pubsub_pkg/src/talker_custom.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "custom_interfaces/msg/location.hpp"

using namespace std::chrono_literals;

/* This example creates a subclass of Node and uses std::bind() to register a
* member function as a callback from the timer. */

class Talker : public rclcpp::Node
{
  public:
    Talker()
    : Node("custom_talker"), count_(0)
    {
      publisher_ = this->create_publisher<custom_interfaces::msg::Location>("custom_topic", 10);
      timer_ = this->create_wall_timer(
      500ms, std::bind(&Talker::timer_callback, this));
    }

  private:
    void timer_callback()
    {
      auto msg = custom_interfaces::msg::Location();
      msg.location.x = 1;
      msg.location.y = 1;
      msg.location.z = 0;
      msg.name = "honolulu";

      RCLCPP_INFO(this->get_logger(), "Publishing: (%f, %f, %f)/%s", 
                    msg.location.x, msg.location.y, 
                    msg.location.z, msg.name.c_str());
      publisher_->publish(msg);
    }
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<custom_interfaces::msg::Location>::SharedPtr publisher_;
    size_t count_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<Talker>());
  rclcpp::shutdown();
  return 0;
}

File: src/py_pubsub_pkg/py_pubsub_pkg/talker_custom.py
import rclpy
from rclpy.node import Node
from custom_interfaces.msg import Location


class CustomTalker(Node):

    def __init__(self):
        super().__init__('custom_talker')
        self.publisher_ = self.create_publisher(Location, 'custom_topic', 10)
        timer_period = 0.5  # seconds
        self.timer = self.create_timer(timer_period, self.timer_callback)
        self.count = 0

    def timer_callback(self):
        msg = Location()
        msg.location.x = 1.0
        msg.location.y = 1.0
        msg.location.z = 0.0
        msg.name = "honolulu"

        self.get_logger().info(
            f'Publishing: ({msg.location.x:.2f}, {msg.location.y:.2f}, {msg.location.z:.2f}) / {msg.name}'
        )
        self.publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = CustomTalker()

    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()